rust/osm2lanes-web/src/main.rs

```diff
@@ -74,8 +55,7 @@ pub enum Msg {
 }
 
 pub struct App {
-    focus_ref: NodeRef,
-    state: State,
+    state: Rc<RefCell<State>>,
```

Comment:
Where was the need for these two introduced? Is it because some of the methods got moved to other modules, and the borrow checker thinks there's a double borrow of two different pieces of `App` at the same time?